File: 2.替换空格.cpp
class Solution {
public:
	void replaceSpace(char *str,int length) {
        int cnt = 0;
        for(int i = 0; i < length; i++)
            if(str[i] == ' ')
                cnt++;
        str = (char*)realloc(str, length + 2 * cnt);
        for(int j = length, i = length + 2 * cnt; i >=0;i--, j--){
            if(str[j] != ' ')
                str[i] = str[j];
            else{
                str[i--] = '0';
                str[i--] = '2';
                str[i] = '%';
            }
        }
	}
};

File: java/2.替换空格.java
public class Solution {
    public String replaceSpace(StringBuffer str) {
    	int cnt = 0;
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) == ' ')
                cnt++;
        }
        int newLen = str.length() + cnt * 2;
        int newIdx = newLen - 1;
        int oldIdx = str.length() - 1;
        str.setLength(newLen);
        for(; oldIdx >=0 && newIdx >= 0; oldIdx--) {
            if(str.charAt(oldIdx) == ' ') {
                str.setCharAt(newIdx--, '0');
                str.setCharAt(newIdx--, '2');
                str.setCharAt(newIdx--, '%');
            }
            else
                str.setCharAt(newIdx--, str.charAt(oldIdx));
        }
        return str.toString();
    }
}
